src/test/java/dev/blaauwendraad/masker/json/ConcurrentMaskingTest.java

```diff
@@ -0,0 +1,85 @@
+package dev.blaauwendraad.masker.json;
+
+import dev.blaauwendraad.masker.json.config.JsonMaskingConfig;
+
+import org.assertj.core.api.Assertions;
+import org.jspecify.annotations.Nullable;
+import org.junit.jupiter.api.Test;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+public class ConcurrentMaskingTest {
+    private static final Duration MAX_CONCURRENT_TEST_RUN = Duration.ofSeconds(10);
+    private static final String JSON_INPUT =
+            """
+        {
+            "userInfo": {
+                "personalInfo": {
+                    "firstName": "Breus",
+                    "lastName": "Blaauwendraad"
+                },
+                "ipAddress": "123.102.123",
+                "userId": 1234
+            }
+        }""";
+
+    private static final String EXPECTED_MASKED_OUTPUT =
+            """
+        {
+            "userInfo": {
+                "personalInfo": {
+                    "firstName": "***",
+                    "lastName": "***"
+                },
+                "ipAddress": "***",
+                "userId": 1234
+            }
+        }""";
+
+    @Test
+    void concurrentJsonPathMasking() throws ExecutionException, InterruptedException {
+        JsonMasker masker =
+                JsonMasker.getMasker(
+                        JsonMaskingConfig.builder()
+                                .maskJsonPaths("$.userInfo.personalInfo", "$.userInfo.ipAddress")
+                                .build());
+        maskConcurrently(Runtime.getRuntime().availableProcessors(), masker);
+    }
+
+    @Test
+    void concurrentKeyMasking() throws ExecutionException, InterruptedException {
+        JsonMasker masker =
+                JsonMasker.getMasker(
+                        JsonMaskingConfig.builder().maskKeys("firstName", "lastName", "ipAddress").build());
+        maskConcurrently(Runtime.getRuntime().availableProcessors(), masker);
+    }
+
+    private void maskConcurrently(int nThreads, JsonMasker masker) throws ExecutionException, InterruptedException {
+        ExecutorService executorService = Executors.newFixedThreadPool(nThreads);
+        List<Future<@Nullable String>> futures = new ArrayList<>();
```

Comment:
@Breus could've used a `var` to hide your nullable war crimes...

Reply:
😂😂😂😂